base/intfuncs.jl

```diff
@@ -198,7 +198,7 @@ end
 function ndigits0z(n::Unsigned, b::Int)
     d = 0
     if b < 0
-        d = ndigits0znb(n, b)
+        d = ndigits0znb(signed(n), b)
```

Comment:
This change actually cancels the one I incorrectly made in a previous commit. So after squashing nothing changes here.